src/napi-inl.h

```diff
@@ -0,0 +1,1724 @@
+  if (status != napi_ok) throw Error::New(Env());
+  return result;
+}
+
+inline Persistent Value::MakePersistent() const {
+  napi_persistent persistent;
+  napi_status status = napi_create_persistent(_env, _value, &persistent);
+  if (status != napi_ok) throw Error::New(Env());
+  return Persistent(_env, persistent);
+}
+
+inline Env Value::Env() const {
+  return Napi::Env(_env);
+}
+
+inline napi_valuetype Value::Type() const {
+  if (_value == nullptr) {
+    return napi_undefined;
+  }
+
+  napi_valuetype type;
+  napi_status status = napi_get_type_of_value(_env, _value, &type);
+  if (status != napi_ok) throw Error::New(Env());
+  return type;
+}
+
+inline bool Value::IsUndefined() const {
+  return Type() == napi_undefined;
+}
+
+inline bool Value::IsNull() const {
+  return Type() == napi_null;
+}
+
+inline bool Value::IsBoolean() const {
+  return Type() == napi_boolean;
+}
+
+inline bool Value::IsNumber() const {
+  return Type() == napi_number;
+}
+
+inline bool Value::IsString() const {
+  return Type() == napi_string;
+}
+
+inline bool Value::IsSymbol() const {
+  return Type() == napi_symbol;
+}
+
+inline bool Value::IsArray() const {
+  if (_value == nullptr) {
+    return false;
+  }
+
+  bool result;
+  napi_status status = napi_is_array(_env, _value, &result);
+  if (status != napi_ok) throw Error::New(Env());
+  return result;
+}
+
+inline bool Value::IsArrayBuffer() const {
+  if (_value == nullptr) {
+    return false;
+  }
+
+  bool result;
+  napi_status status = napi_is_arraybuffer(_env, _value, &result);
+  if (status != napi_ok) throw Error::New(Env());
+  return result;
+}
+
+inline bool Value::IsTypedArray() const {
+  if (_value == nullptr) {
+    return false;
+  }
+
+  bool result;
+  napi_status status = napi_is_typedarray(_env, _value, &result);
+  if (status != napi_ok) throw Error::New(Env());
+  return result;
+}
+
+inline bool Value::IsObject() const {
+  return Type() == napi_object;
+}
+
+inline bool Value::IsFunction() const {
+  return Type() == napi_function;
+}
+
+inline bool Value::IsBuffer() const {
+  if (_value == nullptr) {
+    return false;
+  }
+
+  bool result;
+  napi_status status = napi_buffer_has_instance(_env, _value, &result);
+  if (status != napi_ok) throw Error::New(Env());
+  return result;
+}
+
+inline Boolean Value::AsBoolean() const {
+  return Boolean(_env, _value);
+}
+
+inline Number Value::AsNumber() const {
+  return Number(_env, _value);
+}
+
+inline String Value::AsString() const {
+  return String(_env, _value);
+}
+
+inline Object Value::AsObject() const {
+  return Object(_env, _value);
+}
+
+inline Array Value::AsArray() const {
+  return Array(_env, _value);
+}
+
+inline TypedArray Value::AsTypedArray() const {
+  return TypedArray(_env, _value);
+}
+
+inline Function Value::AsFunction() const {
+  return Function(_env, _value);
+}
+
+inline Buffer Value::AsBuffer() const {
+  return Buffer(_env, _value);
+}
```

Comment:
No good reason. It would probably be better to use As<T> to be more similar to V8.